client/scripts/views/pages/profile/profile_header.ts

```diff
@@ -57,20 +67,60 @@ const editIdentityAction = (account, currentIdentity: SubstrateIdentity, vnode)
 
 export interface IProfileHeaderAttrs {
   account;
+  showJoinCommunityButton: boolean;
   refreshCallback: Function;
 }
 
 export interface IProfileHeaderState {
+  onLinkedProfile: boolean;
   subscription: Unsubscribable | null;
   identity: SubstrateIdentity | null;
   copied: boolean;
+  loading: boolean;
 }
 
 const ProfileHeader: m.Component<IProfileHeaderAttrs, IProfileHeaderState> = {
   view: (vnode) => {
     const { account, refreshCallback } = vnode.attrs;
-    const onOwnProfile = account.chain === app.user.activeAccount?.chain?.id
-      && account.address === app.user.activeAccount?.address;
+    const onOwnProfile = typeof app.user.activeAccount?.chain === 'string'
+      ? (account.chain === app.user.activeAccount?.chain && account.address === app.user.activeAccount?.address)
+      : (account.chain === app.user.activeAccount?.chain?.id && account.address === app.user.activeAccount?.address);
+
+    const showJoinCommunityButton = vnode.attrs.showJoinCommunityButton && !onOwnProfile;
+
+    if (app.user.activeAccounts.length === 0) {
+      return m(PageLoading);
+    }
+
+    const onLinkedProfile = !onOwnProfile && app.user.activeAccounts.filter((account_) => {
+      return app.user.getRoleInCommunity({
+        account: account_,
+        chain: app.activeChainId(),
+      });
+    }).filter((account_) => {
+      return account_.address === account.address;
+    }).length > 0;
+
+    const joinCommunity = async () => {
+      if (!app.activeChainId() || onOwnProfile) return;
+      vnode.state.loading = true;
+      const addressInfo = app.user.addresses
+        .find((a) => a.address === account.address && a.chain === app.activeChainId());
+      try {
+        await app.user.createRole({
+          address: addressInfo,
+          chain: app.activeChainId(),
+          community: app.activeCommunityId(),
+        });
+        vnode.state.loading = false;
+        setActiveAccount(account);
```

Comment:
Missing await statement here?

Reply:
I was going on precedent—everywhere else in the codebase, setActiveAccount is called without an await. Happy to add it though, I also did a double-take when I saw, but thought it best to follow precedent.

Reply:
If that's the case, the other instances need to be updated (recommend doing so in a different PR). It's probably because setActiveAccount used to be synchronous. But here, the missing await means the m.redraw() won't function as expected.